prediction_market_agent/omen.py

```diff
@@ -275,9 +400,69 @@ def binary_omen_buy_outcome_tx(
     )
 
 
-def omen_sell_outcome_tx(market: Market) -> None:
-    # TODO in next MR.
-    ...
+def omen_sell_outcome_tx(
+    amount: xDai,
+    from_address: HexAddress,
+    from_private_key: PrivateKey,
+    market: Market,
+    outcome: str,
+    auto_withdraw: bool,
+) -> None:
+    """
+    Sells the given amount of shares for the given outcome in the given market.
+    """
+    web3 = Web3(Web3.HTTPProvider(GNOSIS_RPC_URL))
+    amount_wei = xdai_to_wei(amount)
+
+    # Get the index of the outcome we want to buy.
+    outcome_index: int = market.get_outcome_index(outcome)
+
+    # Get the current nonce for the given from_address.
+    # If making multiple transactions quickly after each other,
+    # it's better to increae it manually (otherwise we could get stale value from the network and error out).
+    nonce = web3.eth.get_transaction_count(from_address)
+
+    # Calculate the amount of shares we will sell for the given selling amount of xdai.
+    max_outcome_tokens_to_sell = omen_calculate_sell_amount(
+        web3, market, amount_wei, outcome_index
+    )
+
+    # Approve the market maker to move our (all) conditional tokens.
+    approve_tx_receipt = omen_approve_all_market_maker_to_move_conditionaltokens_tx(
+        web3=web3,
+        market=market,
+        approve=True,
+        from_address=from_address,
+        from_private_key=from_private_key,
+        tx_params={"nonce": nonce},
+    )
+    nonce += ONE_NONCE  # Increase after each tx.
+    check_tx_receipt(approve_tx_receipt)
```

Comment:
Do you know what happens if you call `omen_sell_outcome_tx` on a market that has closed? Will this this check fail?

Reply:
It will not fail here, but it will fail while calling the contract, on something like `ContractLogicError: execution reverted: SafeMath: subtraction overflow`. I read that `receipt["status"]` can be zero (meaning there was some error), but so far all the errors I had were during the contract calling, not afterward from `check_tx_receipt`. But I add this anyway just to be sure.